### PR TITLE
496 endpoint to check for existent rule

### LINKED ISSUE #496: Endpoint to check for existent rule
### Description
Check if a rule with a given topic and expression already exists (in service of editing rules, or creating new ones, to avoid creating duplicate rules and for convenience if the client ever wants to check if a rule exists still server side)

Related diagram: 
```mermaid
sequenceDiagram
    participant User
    participant Client
    participant Server
    
    User->>Client: Click Add button
    
    Note over Client,Server: (a) Check for Existing Rule
    Client->>Server: POST /rules/check (topic, expr)
    Note right of Server: Check if rule exists in rules hashmap
    Server-->>Client: Return rule_id or null
    
    Note over Client: (b) User Confirmation Flow
    alt Matching rule exists
        Client->>User: Ask: Sure you want to make rule again?
        User->>Client: Confirm
    else No matching rule
        Note over Client: Continue to add
    end
    
    Note over Client,Server: (c) Add Rule
    Client->>Server: POST /rules/add (Rule, client_id)
    Note right of Server: (d) Server Logic: Insert (client, rule_id) into BiMap, Add rule_id to topic_index, Insert rule into rules hashmap
    Server->>Server: Add rule
    Server-->>Client: Success response
    
    Note over Client,User: (e) UI Update
    Client->>User: Update rule list display
```

### Acceptance Criteria
- Returns existing rule_id if found
- Returns null/404 if not found

### Proposed Solution
```
POST /rules/check
Body: { topic: Topic, expr: String }
Response: { rule_id?: RuleId }
```
Add `check_rule()` method to RuleManager in `rule_structs.rs`, add endpoint to `rule_controller.rs`. Reference `get_all_rules()` for hashmap read pattern. Test using Postman for basic functionality, then add tests to `rule_structs_test.rs`.

### Mocks
_No response_

### PR DESCRIPTION
**Pull request by bracyw**
## Changes

_Explanation of changes goes here_

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `package-lock.json` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: scylla-server/src/controllers/rule_controller.rs [hunk 1]
```diff
@@ -72,3 +72,12 @@ pub async fn get_all_rules_with_client_info(
             .await,
     ))
 }
+
+#[debug_handler]
+pub async fn check_rule(
+    Extension(rules_manager): Extension<Arc<RuleManager>>,
+    Json(rule): Json<Rule>,
+) -> Json<bool> {
+    debug!("Checking if rule exists: {} - {}", rule.topic, rule.expr());
```
Comment (bracyw) [inline scylla-server/src/controllers/rule_controller.rs]: remove

File: scylla-server/src/rule_structs.rs [hunk 2]
```diff
@@ -247,6 +247,10 @@ impl Rule {
         }
     }
 
+    pub fn expr(&self) -> &str {
```
Comment (bracyw) [inline scylla-server/src/rule_structs.rs]: probably don't need this, if not justify
